### PR TITLE
Update master branch from Train to Xena

### PR DESCRIPTION
**Pull request by msherman64**
This PR contains the needed changes to move us from Train to Xena on the master branch. It also contains a number of bug-fixes and convenience features, some of which could be backported to train.

Backportable changes:
- Defaults for allowing a minimal dev site (disable tls, disable federation, set ironic inspection network and site_name)
- ensure that variables target the deploy host, in particular the path for templating the admin-openrc.sh
- split post-deploy into roles and playbooks for easier maintenance
- Allow deployment of KVM compute node

Breaking Changes:
New passwords must be set, use the `kolla-mergepwd` tool to pull in new values from the site-config.example

The hosts file has major changes, in particular the replacement of the HAProxy with Loadbalancer groups, addition of "common" group mostly replacing baremetal, and addition of a "defaults" group for the deploy host fix above. Back up your ansible hosts file, and then copy the one from site-config.example into its place, and update the primary groups as usual. Take care to review any customizations you may have done, and see if they are still needed.

### CODE REVIEW

File: kolla/defaults.yml [hunk 1]
```diff
@@ -196,7 +205,7 @@ enable_jupyterhub_external_frontend: "{{ enable_jupyterhub_external }}"
 # Neutron
 enable_neutron: yes
 enable_openvswitch: yes
-enable_neutron_fwaas: yes
+enable_neutron_fwaas: no
```
Comment (Mark-Powers) [inline kolla/defaults.yml:210]: is `fwaas` supported anymore? I don't think it has a xena release. If not, can this be removed entirely?

File: kolla/defaults.yml [hunk 2]
```diff
@@ -129,18 +130,22 @@ horizon_regions:
     blazar_database_password: "{{ blazar_database_password }}"
 
 # Keystone IdP federation
+identity_provider_key: "{{ 'EABNO4wxhX0ZOryErTXnWPUkCAcWxTFHwZ6f_zjDHLg' if keycloak_hostname == 'auth.dev.chameleoncloud.org' else '02GplnOEWB4gwqrZDyyiaNdIrhgdinNnObXcPHy2RTo' }}"
```
Comment (Mark-Powers) [inline kolla/defaults.yml:133]: Is it possible to convert these to friendlier names?

Comment (Mark-Powers) [inline kolla/defaults.yml:133]: Is it possible to convert these to friendlier names?

**Review by Mark-Powers — APPROVED** [review]

File: kolla/defaults.yml [hunk 3]
```diff
@@ -216,7 +225,7 @@ neutron_dev_plugins:
 # Nova
 enable_nova: yes
 enable_nova_serialconsole_proxy: yes
-nova_compute_virt_type: none
+nova_compute_virt_type: kvm
```
Comment (codyhammock) [inline kolla/defaults.yml:228]: Should this be the default value for a baremetal server?

**Comment by msherman64** [comment]
This is on hold until all associate sites are upgraded, as changing what the "master" branch does will affect existing sites.

**Pull request closed without merging**